jmespath/api.go

```diff
@@ -0,0 +1,24 @@
+package jmespath
+
+// NewLexer creates a new JMESPath lexer.
+func NewLexer() *Lexer {
```

Comment:
I'd suggest these published functions `NewLexer` and `NewParser` to be in their associated lexer.go and paser.go.  Users will tend to be looking at docs on godoc like, http://godoc.org/github.com/jmespath/jmespath.go/jmespath

When they do look at the code it is generally helpful to have the helper  (constructor) in the same file as the struct they initialize.